pkg/controller/redissnapshot/redissnapshot_controller.go

```diff
@@ -0,0 +1,246 @@
+	err = c.Watch(&source.Kind{Type: &integreatlyv1alpha1.RedisSnapshot{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource Pods and requeue the owner RedisSnapshot
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &integreatlyv1alpha1.RedisSnapshot{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileRedisSnapshot implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileRedisSnapshot{}
+
+// ReconcileRedisSnapshot reconciles a RedisSnapshot object
+type ReconcileRedisSnapshot struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client            client.Client
+	scheme            *runtime.Scheme
+	logger            *logrus.Entry
+	ConfigManager     croAws.ConfigManager
+	CredentialManager croAws.CredentialManager
+}
+
+// Reconcile reads that state of the cluster for a RedisSnapshot object and makes changes based on the state read
+// and what is in the RedisSnapshot.Spec
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileRedisSnapshot) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.logger.Info("reconciling redis snapshot")
+	ctx := context.TODO()
+
+	// Fetch the RedisSnapshot instance
+	instance := &integreatlyv1alpha1.RedisSnapshot{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// check status, if complete return
+	if instance.Status.Phase == croType.PhaseComplete {
+		r.logger.Infof("snapshot for %s exists", instance.Name)
+		return reconcile.Result{}, nil
+	}
+
+	// get redis cr
+	redisCr := &integreatlyv1alpha1.Redis{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: instance.Spec.ResourceName, Namespace: instance.Namespace}, redisCr)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to get redis cr : %s", err.Error())
+		if updateErr := resources.UpdateSnapshotPhase(ctx, r.client, instance, croType.PhaseFailed, croType.StatusMessage(errMsg)); updateErr != nil {
+			return reconcile.Result{}, updateErr
+		}
+	}
+
+	// check redis cr deployment type is aws
+	if redisCr.Status.Strategy != providers.AWSDeploymentStrategy {
+		errMsg := "none supported deployment strategy"
```

Comment:
This message is difficult to understand. Maybe something like:

`the resource <resourceName> uses an unsupported provider strategy <strategy>, only resources using the aws provider are valid`